File: examples/scala/src/main/scala/com/xebia/functional/xef/scala/auto/Love.scala
package com.xebia.functional.xef.scala.auto

import com.xebia.functional.xef.scala.auto.*
import io.circe.Decoder
import com.xebia.functional.xef.prompt.Prompt

private final case class Love(emojis: List[String]) derives SerialDescriptor, Decoder

@main def runLove: Unit =
  conversation {
    val love = prompt[Love](Prompt("Tell me you like me with just emojis"))
    println(love.emojis.mkString(""))
  }

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/Love.kt
package com.xebia.functional.xef.auto

import com.xebia.functional.xef.auto.llm.openai.OpenAI
import com.xebia.functional.xef.auto.llm.openai.prompt
import com.xebia.functional.xef.auto.llm.openai.promptMessage
import com.xebia.functional.xef.prompt.Prompt

suspend fun main() {
  OpenAI.conversation {
    val love: String = prompt(Prompt("tell me you like me with just emojis"))
    println(love)
  }
}
